dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataFileServiceBean.java

```diff
@@ -1200,6 +1206,21 @@ public boolean isSameTermsOfUse(FileTermsOfUse termsOfUse1, FileTermsOfUse terms
         } else {
             throw new IOException("Temp directory is not configured.");
         }
+        
+        if (settingsService.isTrueForKey(SettingsServiceBean.Key.AntivirusScannerEnabled) &&
+                tempFile.toFile().length() <= settingsService.getValueForKeyAsLong(SettingsServiceBean.Key.AntivirusScannerMaxFileSize)) {
+
+            String scannerMessage = fileService.scan(tempFile.toFile());
+
+            if (scannerMessage.contains("FOUND")) {
+                logger.warn("There was an attempt to upload file infected with virus. Scanner message: {} Dataset version: {}", scannerMessage, version);
+                if (!tempFile.toFile().delete()) {
```

Comment:
We should use .nio when we can since it is newer and more efficient implementation. So I would advise to use: `(!Files.deleteIfExists(tempFile))`

Reply:
I already know that file exists. So there is no point of `if` in this case. If we use `Files.delete(tempFile)` then we would have to think about it.
Key difference is: do we want to throw IOException or not.
If we would use `Files.delete(tempFile)` then in case of failure:
 - when uploading from ui (EditDataFilesPage) - we would have no info for the user about some failure - it would silently skipped from user perspective (only one extra entry in log)
 - when uploading from api - we would have message: "There was an error when trying to add the new file." and then probably some info about file could not be deleted, maybe with path to this file.

Maybe in case of api it would be enough (but personally I would probably not expose message from IOException - it can show to much of server internals)
But in case of ui I would argue that `File.delete` is better choice (at least for now).

What do you think about it?

Reply:
So I understand that now if the file is not deleted there will be info on UI since the error is not in Try-catch? If that's the case then I understand it would require bigger refactoring so I would leave it as it is.

Reply:
There is try-catch but it doesn't affect what is presented to the user:
```
            try {
                // Note: A single uploaded file may produce multiple datafiles - 
                // for example, multiple files can be extracted from an uncompressed
                // zip file. 
                dFileList = datafileDao.createDataFiles(workingVersion, uFile.getInputStream(), uFile.getFileName(), uFile.getContentType());

            } catch (IOException ioex) {
                logger.warning("Failed to process and/or save the file " + uFile.getFileName() + "; " + ioex.getMessage());
                return;
            } ...
```

Reply:
Yeah that's what I meant, just a mental shortcut, sorry :P.